modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -859,7 +885,17 @@ public CompletableFuture<?> onCreate(ConfigurationNotificationEvent<Distribution
 
             saveDataNodesAndUpdateTriggerKeysInMetaStorage(zoneId, ctx.storageRevision(), logicalTopology);
 
-            return completedFuture(null);
+            return zonesByIdVv.update(ctx.storageRevision(), (zones, e) -> {
+                if (e != null) {
```

Comment:
What about busy lock? Do we need it here?

Reply:
Fixed. I removed this code.